packs/excel/actions/lib/excel_reader.py

```diff
@@ -0,0 +1,239 @@
+"""
+Copyright 2016 Brocade Communications Systems, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from openpyxl import Workbook, load_workbook
+import os
+import time
+
+
+class ExcelReader(object):
+    def __init__(self, excel_file, lock=False, lock_retries=3, lock_delay=1):
+        ''' Loads the excel configuration file '''
+        self._filename = excel_file
+        self._lock = lock
+
+        # Set class variables
+        self._strict = True
+        self._wb = None
+        self._ws = None
+        self._sheet_name = None
+        self._key_column = -1
+        self._keys = []
+        self._var_name_row = -1
+        self._variable_start_column = -1
+        self._variable_end_column = -1
+        self._data_start_row = -1
+        self._data_end_row = -1
+
+        if lock:
+            # is file already locked?
+            attempts = 0
+            while os.path.isfile("%s.lock" % excel_file):
+                time.sleep(lock_delay)
+                attempts += 1
+                if attempts >= lock_retries:
+                    raise IOError("File is locked")
+            # it's now ours, so lock it
+            try:
+                open("%s.lock" % excel_file, 'a').close()
```

Comment:
Ah, you are right - might be worth looking in the other two suggested packages although StackStorm code and a lot of other projects I know still use `lockfile` so it's probably also not a big deal if we still use it :)